src/main/java/bio/terra/common/FutureUtils.java

```diff
@@ -31,56 +31,52 @@ public static <T> List<T> waitFor(final List<Future<T>> futures) {
    * @return The resolved values of the futures. There is no guarantee that the order of returned
    *     values matches the order of the passed in futures
    */
-  public static <T> List<T> waitFor(
-      final List<Future<T>> futures, final Optional<Duration> maxThreadWait) {
+  public static <T> List<T> waitFor(final List<Future<T>> futures, final Duration maxThreadWait) {
 
-    final AtomicReference<Optional<ApiException>> foundFailure =
-        new AtomicReference<>(Optional.empty());
+    final AtomicReference<ErrorReportException> foundFailure = new AtomicReference<>();
     try (Stream<Future<T>> stream = futures.stream()) {
```

Comment:
I seem to recall mentioning this before, but there's no need to `stream()` inside a `try()` here as the stream returned by `Collection.stream()` doesn't need to be closed.